internal/analytics/client/sync/client.go

```diff
@@ -212,10 +212,14 @@ func (a *Client) EventWithSourceAndLabel(category, action, source, label string,
 	}
 
 	a.customDimensions.Sequence = ptr.To(a.sequence)
-	a.sequence++
 
 	actualDims := mergeDimensions(a.customDimensions, dims...)
 
+	if a.sequence == *actualDims.Sequence {
+		// Increment the sequence number unless dims overrides it (e.g. heartbeats use -1).
+		a.sequence++
+	}
```

Comment:
This is needed to pass https://github.com/ActiveState/cli/blob/3f28923089bf7c2ad43b0ab806631d64162684e3/test/integration/analytics_int_test.go#L208. Heartbeat events with a `-1` sequence were needlessly triggering an increment of `a.sequence` here, resulting in seemingly non-sequential events.

Reply:
Nice catch!